ctms/acoustic_service.py

```diff
@@ -396,16 +396,24 @@ def _add_contact(self, list_id=None, sync_fields=None, columns=None):
                 self.context["main_status"] = status
                 self.context["main_duration_s"] = duration_s
 
-    def _insert_update_relational_table(self, table_name, rows):
-        if not rows:
-            return
+    def _replace_relational_table_data(self, table_name, email_id, rows):
         start_time = time.monotonic()
-        status = "success"
         table_id = self.relational_tables[table_name]
+        status = "success"
+        method = "delete_relational_table_data"
         try:
-            self.acoustic.insert_update_relational_table(
-                table_id=table_id, rows=rows
-            )  # Call to Acoustic
+            # Calls to Acoustic
+            self.acoustic.delete_relational_table_data(
+                table_id=table_id,
+                rows=[
+                    {"email_id": email_id},
+                ],
+            )
+            if rows:
+                method = "insert_update_relational_table"
+                self.acoustic.insert_update_relational_table(
+                    table_id=table_id, rows=rows
+                )
```

Comment:
What happens if something goes wrong between deleting all rows and re-adding them? Will we attempt to retry?

Can we attempt to say, only delete the RT rows for unsubscribed newsletters? I think we might have to do this anyway, since [the docs say](https://developer.goacoustic.com/acoustic-campaign/reference/deleterelationaltabledata):

> If the table has multiple key columns, each ROW has multiple COLUMN elements. All key columns must be provided for each ROW.

Looking at the Acoustic newsletters RT schema, both `email_id` and `newsletter_name` have the little key icon in the UID column
<img width="237" alt="image" src="https://user-images.githubusercontent.com/20747904/223511676-1d2d4a53-5b7d-4d2a-be62-8502bb039642.png">

So unless you tried this and it worked already, we might have to provide the newsletter name for each row we want to delete anyway.

Reply:
You're right! 

This PR was a rabbit hole